go/types.go

```diff
@@ -179,10 +189,51 @@ type TBSCertificateListWithRawSerials struct {
 	RevokedCertificates []RevokedCertificateWithRawSerial `asn1:"optional"`
 }
 
+type Extension struct {
+	Id       asn1.ObjectIdentifier
+	Critical bool `asn1:"optional"`
+	Value    []byte
+}
+
 type RevokedCertificateWithRawSerial struct {
 	Raw            asn1.RawContent
 	SerialNumber   asn1.RawValue
 	RevocationTime time.Time
+	Extensions     []Extension `asn1:"optional"`
+}
+
+func (c RevokedCertificateWithRawSerial) Reason() (asn1.Enumerated, error) {
+	seen := false
+	reasonCode := asn1.Enumerated(0)
+	for _, ext := range c.Extensions {
+		if ext.Id.Equal(kOidExtensionReasonCode) {
+			if seen {
+				return reasonCode, fmt.Errorf("Repeated CRLReason extension")
+			}
+			_, err := asn1.Unmarshal(ext.Value, &reasonCode)
+			if err != nil {
+				return reasonCode, err
+			}
+			seen = true
+		}
+	}
+	return reasonCode, nil
+}
+
+func (c RevokedCertificateWithRawSerial) SerialAndReason() (SerialAndReason, error) {
+	reason, err := c.Reason()
+	if err != nil {
+		return SerialAndReason{}, err
+	}
+
+	if reason < 0 || reason > 255 {
```

Comment:
Do we have a test for this?

Reply:
Good catch. I should have had this in `Reason()` instead of `SerialAndReason()`. I added some tests.